<?xml version="1.0"?>
<robot name="laser_scanner">

<!-- The base_link frame is the actual pose of the laser_scanner -->
  <link name="base_link" />

<!--The following give the static transforms of the various scanner components relative to the
    base_link-->
  <link name="laser_lsl" />
  <link name="laser_lsm" />
  <link name="corrected_imu" />
  <link name="cam" />


  <joint name="base_to_lsl" type="fixed">
    <parent link="base_link"/>
    <child link="laser_lsl"/>
    <origin xyz="-0.035 0 0.085" rpy="0 0 0" />
  </joint>

  <joint name="base_to_lsm" type="fixed">
    <parent link="base_link"/>
    <child link="laser_lsm"/>
    <origin xyz="-0.035 -0.085 0" rpy="-1.57 0 -1.57" />
  </joint>

  <joint name="base_to_imu" type="fixed">
    <parent link="base_link"/>
    <child link="corrected_imu"/>
    <origin xyz="-0.08 0.065 0.03" rpy="3.1415 0 3.1415" />
  </joint>

  <joint name="base_to_cam" type="fixed">
    <parent link="base_link"/>
    <child link="cam"/>
    <origin xyz="0 0.065 0" rpy="0 0 0" />
  </joint>

</robot>
